src/com/google/javascript/jscomp/DefaultPassConfig.java

```diff
@@ -261,6 +262,11 @@ private void addModuleRewritingPasses(List<PassFactory> checks, CompilerOptions
       checks.add(rewriteCommonJsModules);
     }
 
+    if (options.getLanguageIn().toFeatureSet().contains(Feature.DYNAMIC_IMPORT)) {
+      // Temporary location until this pass does more than warn about usage
+      checks.add(es2020RewriteDynamicImport);
```

Comment:
Instead of being added here, the pass should be added in `addModuleRewritingPasses()`.

Also, it should be added by calling a new method `TranspilationPasses.addDynamicImportPass(checks)`.
We want to keep the transpilation pass creation in the `TranspilationPasses` class.

Reply:
See `TranspliationPasses.addEs6ModulePass()` as an example.

Reply:
Sorry, scratch that.

It's important that the dynamic import still be in the AST when type checking runs so we have a chance to do the type checking we want there.

The pass should get added at the start of `TranspilationPasses.addPostCheckTranspilationPasses()`.

Reply:
Moved